ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -618,14 +819,16 @@ private void verifyState(boolean isLogSegmented) throws IOException, StoreExcept
       TreeMap<MockId, IndexValue> referenceIndexSegment = referenceIndexEntry.getValue();
       IndexSegment realIndexSegment = realIndexEntry.getValue();
       List<MessageInfo> messageInfos = new ArrayList<>();
-      FindEntriesCondition condition = new FindEntriesCondition(Long.MAX_VALUE);
       assertTrue("There should have been entries returned from the index segment",
-          realIndexSegment.getEntriesSince(null, condition, messageInfos, new AtomicLong(0)));
+          realIndexSegment.getEntriesSince(null, new FindEntriesCondition(Long.MAX_VALUE), messageInfos,
+              new AtomicLong(0)));
       assertEquals("Size of index segment differs from expected", referenceIndexSegment.size(), messageInfos.size());
       for (Map.Entry<MockId, IndexValue> referenceIndexSegmentEntry : referenceIndexSegment.entrySet()) {
         IndexValue value = realIndexSegment.find(referenceIndexSegmentEntry.getKey());
         IndexValue referenceValue = referenceIndexSegmentEntry.getValue();
-        assertEquals("Value from IndexSegment does not match expected", referenceValue.getBytes(), value.getBytes());
+        if (index.hardDeleter.enabled.get() && !deletedKeys.contains(referenceIndexSegmentEntry.getKey())) {
```

Comment:
For delete record, operationTime is expected to be -1 right. So, the referenceIndex and the actual IndexShould have the same operationTime. I guess both values should be equivalent irrespective of whether hard delete is enabled or not. Or am I missing something